### PR TITLE
Implemented some interesting user suggestions

### PR DESCRIPTION
**Pull request by alexanderkind**
Implemented some interesting user suggestions:
1) Rounding data module and eye corners #192 
2) Rounding data module outside corners
3) Background shape behind embedded image with rounding corners (EmbeddedImageShape)
4) Adjusting the cutout of an embedded image #190 
5) Gradient for all (QrDataModuleShape, QrEyeShape, EmbeddedImageShape) #143 

<p float="left">

</p>
<p float="left">

</p>

### CODE REVIEW

**Comment by vanyasem** [comment]
do you really have to import `package:flutter/material.dart` in `types.dart` just to use `Colors.black`?

File: example/assets/images/4.0x/logo_yakka_transparent.png
Comment (vanyasem) [inline example/assets/images/4.0x/logo_yakka_transparent.png:1]: What is Yakka? Why it's logo is used as an example?

File: example/lib/main_screen.dart [hunk 1]
```diff
@@ -24,7 +24,7 @@ class _MainScreenState extends State<MainScreen> {
         // ignore: lines_longer_than_80_chars
         'Hey this is a QR code. Change this value in the main_screen.dart file.';
 
-    final qrFutureBuilder = FutureBuilder<ui.Image>(
+    /*final qrFutureBuilder = FutureBuilder<ui.Image>(
```
Comment (vanyasem) [inline example/lib/main_screen.dart:27]: Remove commented code?

File: lib/src/qr_painter.dart [hunk 2]
```diff
@@ -218,49 +276,117 @@ class QrPainter extends CustomPainter {
...
-          canvas.drawRRect(roundedRect, paint);
+        final squareRect = _createDataModuleRect(paintMetrics, x, y, gap);
+        // check safeArea
+        if(embeddedImageStyle.safeArea
+            && safeAreaRect?.overlaps(squareRect) == true) continue;
+        switch(dataModuleStyle.dataModuleShape) {
+          case QrDataModuleShape.square:
+            if(dataModuleStyle.borderRadius > 0) {
+
+              // If pixel isDart == true and outside safe area
```
Comment (vanyasem) [inline lib/src/qr_painter.dart:287]: typo `isDart` -> `isDark`

File: example/lib/main_screen.dart (hunk 1 above)
Comment (alexanderkind) [reply example/lib/main_screen.dart:23]: Left for an example on preloading an image

**Comment by vanyasem** [comment]
thank you for addressing the review so quickly!

**Comment by razfazz** [comment]
wow very nice! thank you

**Comment by mminhlequang** [comment]
up

**Comment by 2Math0** [comment]
up

**Comment by benjdero** [comment]
@vanyasem What's preventing this from being merged? The proper image cutout alone makes the result look much more professional.

**Comment by martipello** [comment]
LGTM!!!

**Comment by Hilbert2048** [comment]
@vanyasem why this update is still not merged?

**Comment by vanyasem** [comment]
@Hilbert2048 because I am just a random contributor, and not a maintainer of this project. I do not have write access to this repository, and thus I cannot merge this PR :)
You need to contact the actual maintainers of this repo by email: hello@theyakka.com

**Comment by Hilbert2048** [comment]
I think @lukef might have the write permission, would you please have a look at this pull request?

**Comment by guo-hang** [comment]
If the background is black, the interface display is abnormal

**Comment by lukef** [comment]
Please open a separate issue

**Pull request merged**